handler/interceptors.go

```diff
@@ -103,13 +103,28 @@ func (interceptor *paymentValidationInterceptor) intercept(srv interface{}, ss g
 	}
 	log.Debug("Payment validated")
 
-	e := handler(srv, ss)
+	handlerSucceed := false
+
+	defer func() {
```

Comment:
I would suggest moving ```defer``` call to line 99, as it should also be called in case of panic or error within Validate() method.

Reply:
Updated.